opensearch/src/main/java/org/opensearch/sql/opensearch/executor/OpenSearchQueryManager.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sql.opensearch.executor;
+
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import org.apache.logging.log4j.ThreadContext;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.sql.executor.QueryId;
+import org.opensearch.sql.executor.QueryManager;
+import org.opensearch.sql.executor.execution.AbstractPlan;
+import org.opensearch.threadpool.ThreadPool;
+
+/**
+ * QueryManager implemented in OpenSearch cluster.
+ */
+@RequiredArgsConstructor
+public class OpenSearchQueryManager implements QueryManager {
```

Comment:
Question about the responsibility: what's the different between QueryManager and ExecutionEngine? Is running on caller thread or OS worker thread also part of `OpenSearchExecutionEngine` responsibility?

Reply:
Currently, OpenSearchExecutionEngine execute the physical plan in the same worker thread as OpenSearchQueryManager.

But in future, sql-worker thread poll has shared responsibility
1. OpenSearchQueryManager execute plan in sql-worker thread which make sure the future operation not block NIO thread. 
2. OpenSearchExecutionEngine [should plan the logical plan as execution DAG](https://github.com/penghuo/os-sql/blob/hp/poc/maximus-m1/opensearch/src/main/java/org/opensearch/sql/opensearch/executor/OpenSearchExecutionEngine.java). Then OpenSearchExecutionEngine could reuse worker thread poll to plan task.
Even more, [TaskService](https://github.com/penghuo/os-sql/blob/hp/poc/maximus-m1/opensearch/src/main/java/org/opensearch/sql/opensearch/executor/transport/TransportQLTaskAction.java#L31) could also leverage worker thread pool to execute task.

Reply:
Got it. Thanks!